src/models/tag/stream.rs

```diff
@@ -90,133 +166,157 @@ impl FromIterator<HotTag> for HotTags {
-        let label_slice: &[&str] = &labels;
-
-        let list = Taggers::try_from_multiple_sets(None, label_slice, taggers_limit).await?;
-
-        let hot_tags_stream: HotTags = hot_tags
-            .into_iter()
-            .zip(list)
-            .filter_map(|((label, score), user_ids)| match user_ids {
-                Some((tagger_list, taggers_count)) => {
-                    let taggers_id = Taggers::from_vec(tagger_list);
-                    Some(HotTag {
-                        label,
-                        taggers_id,
-                        post_count: score as u64,
-                        taggers_count,
-                    })
-                }
-                None => None,
-            })
+        let mapping = data.unwrap();
+        // for each value in ranking, look up the value in data
+        let mut hot_tags = Vec::new();
+        for (label, _) in ranking.unwrap() {
+            if let Some(tag) = mapping.0.get(&label) {
+                hot_tags.push(HotTag {
+                    label,
+                    taggers_id: Taggers(
+                        tag.taggers_id
+                            .0
+                            .clone()
+                            .into_iter()
+                            .take(tags_query.taggers_limit)
+                            .collect(),
+                    ),
+                    tagged_count: tag.tagged_count,
+                    taggers_count: tag.taggers_count,
+                });
+            }
+        }
+        Ok(Some(HotTags(hot_tags)))
+    }
+
+    async fn set_to_global_cache(
+        result: HotTags,
+        tags_query: &HotTagsInput,
+    ) -> Result<(), DynError> {
+        let key_parts = HotTags::get_cache_key_parts(tags_query);
+        let key_parts_vector: Vec<&str> =
+            key_parts.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+
+        // turn result which is a vector of HotTag, into a mapping from label to HotTag
+        let mapping = result.clone();
+        let hot_tags_data: HashMap<String, HotTag> = mapping
+            .iter()
+            .cloned()
+            .map(|tag| (tag.label.clone(), tag))
             .collect();
-        Ok(Some(hot_tags_stream))
+
+        // store the data as json in cache
+        HotTagsData::put_index_json(
+            &HotTagsData(hot_tags_data),
+            key_parts_vector.as_slice(),
+            Some(tags_query.timeframe.to_cache_period()),
+        )
+        .await?;
+
+        // store the ranking as sorted set in cache
+        HotTags::put_index_sorted_set(
+            key_parts_vector.as_slice(),
+            result
+                .iter()
+                .map(|tag| (tag.tagged_count as f64, tag.label.as_str()))
+                .collect::<Vec<(f64, &str)>>()
+                .as_slice(),
+            Some(GLOBAL_HOT_TAGS_PREFIX),
+            Some(tags_query.timeframe.to_cache_period()),
+        )
+        .await?;
```

Comment:
Could we do in one loop the creation of the `SORTED_SET` (tag/score) and also, the JSON object (hashMap). Maybe refactor the `HotTags` like that?
```rust
pub struct HotTag {
    taggers_id: Taggers,
    post_count: u64
}
```  
- `label` already is the key of the HashMap, so we do not need that info
- `tagged_count`, we get from the `SORTED_SET` score
- `taggers_count`, we get from the length of the array of `taggers_id`

Once we have the two lists, save in the indexes